src/lib/formats/rpk.cpp

```diff
@@ -0,0 +1,459 @@
+// license:LGPL-2.1+
+// copyright-holders:Michael Zapf
+/***************************************************************************
+
+	rpk.cpp
+
+	RPK format support
+
+	A RPK file ("rompack") contains a collection of dump files and a layout
+	file that defines the kind of circuit board (PCB) used in the cartridge
+	and the mapping of dumps to sockets on the board.
+
+Example:
+	<?xml version="1.0" encoding="utf-8"?>
+	<romset>
+		<resources>
+			<rom id="gromimage" file="ed-assmg.bin" />
+		</resources>
+		<configuration>
+			<pcb type="standard">
+				<socket id="grom_socket" uses="gromimage"/>
+			</pcb>
+		</configuration>
+	</romset>
+
+DTD:
+	<!ELEMENT romset (resources, configuration)>
+	<!ELEMENT resources (rom|ram)+>
+	<!ELEMENT rom EMPTY>
+	<!ELEMENT ram EMPTY>
+	<!ELEMENT configuration (pcb)>
+	<!ELEMENT pcb (socket)+>
+	<!ELEMENT socket EMPTY>
+	<!ATTLIST romset version CDATA #IMPLIED>
+	<!ATTLIST rom id ID #REQUIRED
+	<!ATTLIST rom file CDATA #REQUIRED>
+	<!ATTLIST rom crc CDATA #IMPLIED>
+	<!ATTLIST rom sha1 CDATA #IMPLIED>
+	<!ATTLIST ram id ID #REQUIRED>
+	<!ATTLIST ram type (volatile|persistent) #IMPLIED>
+	<!ATTLIST ram store (internal|external) #IMPLIED>
+	<!ATTLIST ram file CDATA #IMPLIED>
+	<!ATTLIST ram length CDATA #REQUIRED>
+	<!ATTLIST pcb type CDATA #REQUIRED>
+	<!ATTLIST socket id ID #REQUIRED>
+	<!ATTLIST socket uses IDREF #REQUIRED>
+
+***************************************************************************/
+
+#include "rpk.h"
+#include "unzip.h"
+#include "xmlfile.h"
+
+#include <vector>
+
+
+/***************************************************************************
+	TYPE DEFINITIONS
+***************************************************************************/
+
+namespace
+{
+	class rpk_category_impl : public std::error_category
+	{
+	public:
+		virtual char const *name() const noexcept override { return "rpk"; }
+		virtual std::string message(int condition) const override;
+	};
+};
+
+
+/***************************************************************************
+	GLOBAL VARIABLES
+***************************************************************************/
+
+namespace
+{
+	rpk_category_impl const f_rpk_category_instance;
+};
```

Comment:
It’s better to have a single anonymous namespace and keep stuff together rather than having lots of little ones.